.github/workflows/pullrequest.yml

```diff
@@ -4,49 +4,217 @@ on:
   pull_request:
     paths-ignore:
       - 'docs/**'
+      - '**/*.md'
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - 'LICENSE'
+      - '.gitignore'
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
 jobs:
-  build:
-    name: Test and Build on JDK ${{ matrix.java-version }}
+  compile:
+    name: Compile Sources and Tests
     runs-on: ubuntu-latest
-    timeout-minutes: 30
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - java-version: 21
-            sonar-enabled: true
-
+    timeout-minutes: 15
+    outputs:
+      cache-key: ${{ steps.cache-key.outputs.key }}
+    
     steps:
       - name: Checkout code
         uses: actions/checkout@v5
 
-      - name: Set up JDK ${{ matrix.java-version }}
+      - name: Set up JDK 21
```

Comment:
Note that there will come a time that we need to validate with several Java versions again. We need to validate with every LTS, just to provide added certainty to our users. This is why I kept the matrix intact when changing the scripts for AF5.

Reply:
So, pretty curious how that would work with the module break down you have created. Would that mean the number of tasks in the end result explodes to 40? Perhaps we can merge some of those results?